aws-iot-mitigationaction/src/main/java/com/amazonaws/iot/mitigationaction/CreateHandler.java

```diff
@@ -0,0 +1,102 @@
+    private static final int GENERATED_NAME_MAX_LENGTH = 40;
+
+    private final IotClient iotClient;
+
+    public CreateHandler() {
+        iotClient = IotClient.builder().build();
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        CreateMitigationActionRequest createRequest = translateToCreateRequest(request);
+
+        ResourceModel model = request.getDesiredResourceState();
+        if (!StringUtils.isEmpty(model.getMitigationActionArn())) {
+            logger.log(String.format("MitigationActionArn is read-only, but the caller passed %s.",
+                    model.getMitigationActionArn()));
+            // Note: this is necessary even though MitigationActionArn is marked readOnly in the schema.
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "MitigationActionArn is a read-only property and cannot be set.");
+        }
+
+        if (!StringUtils.isEmpty(model.getMitigationActionId())) {
+            logger.log(String.format("MitigationActionId is read-only, but the caller passed %s.",
+                    model.getMitigationActionArn()));
+            // Note: this is necessary even though MitigationActionId is marked readOnly in the schema.
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "MitigationActionId is a read-only property and cannot be set.");
+        }
+
+        CreateMitigationActionResponse createMitigationActionResponse;
+        try {
+            createMitigationActionResponse = proxy.injectCredentialsAndInvokeV2(
+                    createRequest, iotClient::createMitigationAction);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+
+        model.setMitigationActionArn(createMitigationActionResponse.actionArn());
+        model.setMitigationActionId(createMitigationActionResponse.actionId());
+        logger.log(String.format("Created %s with actionId %s.", createMitigationActionResponse.actionArn(),
+                createMitigationActionResponse.actionId()));
+
+        return ProgressEvent.defaultSuccessHandler(model);
+    }
+
+    private static CreateMitigationActionRequest translateToCreateRequest(ResourceHandlerRequest<ResourceModel> request) {
+
+        ResourceModel model = request.getDesiredResourceState();
+
+        // Like most services, we don't require an explicit resource name in the template,
+        // and, if it's not provided, generate one based on the stack ID and logical ID.
+        if (StringUtils.isBlank(model.getActionName())) {
+            // TODO: include stack-id in the generated name. Will do it once the new release of the java plugin
```

Comment:
Let's do this as part of this PR

Reply:
I did already made that change. Missed taking out the TODO. Removed now.